ipn/ipnlocal/c2n.go

```diff
@@ -222,9 +223,11 @@ func (b *LocalBackend) handleC2NPostureIdentityGet(w http.ResponseWriter, r *htt
 
 	res := tailcfg.C2NPostureIdentityResponse{}
 
-	// TODO(kradalby): Use syspolicy + envknob to allow Win registry,
-	// macOS defaults and env to override this setting.
-	if b.Prefs().PostureChecking() {
+	// Only collect serial numbers if enabled on the client,
+	// this will first check syspolicy, MDM settings like Registry
+	// on Windows or defaults on macOS. If they are not set, it falls
+	// back to the cli-flag, `--posture-checking`.
+	if enabled, _ := syspolicy.GetBoolean(syspolicy.PostureChecking, b.Prefs().PostureChecking()); enabled {
```

Comment:
GetBoolean is tested but can you test how your own error handling will work w/ syspolicy?

Reply:
I changed it to log the error and return the default, thanks.

Reply:
I changed it to log the error and return the default, thanks.